src/serde/ser.rs

```diff
@@ -30,9 +93,48 @@ pub fn node_to_stream<W: io::Write>(node: &Node, f: &mut W) -> io::Result<()> {
 }
 
 pub fn node_to_bytes(node: &Node) -> io::Result<Vec<u8>> {
```

Comment:
Maybe can we create a new function that takes a `max_size` parameter? And then we can move this `2_000_000` number higher up in the call stack.